staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -106,10 +107,16 @@ func PatchResource(r rest.Patcher, scope RequestScope, admit admission.Interface
 		)
 
 		userInfo, _ := request.UserFrom(ctx)
-		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		admissionCheck := func(updatedObject runtime.Object, currentObject runtime.Object) error {
-			if mutatingAdmission, ok := admit.(admission.MutationInterface); ok && admit.Handles(admission.Update) {
-				return mutatingAdmission.Admit(admission.NewAttributesRecord(updatedObject, currentObject, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+			if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
+				if mutatingAdmission.Handles(admission.Create) {
+					if accessor, err := meta.Accessor(currentObject); err == nil && accessor.GetResourceVersion() == "" {
```

Comment:
what about `err != nil`? Would rexpect that the error is returned.

Reply:
nit: usually we do `len(x) == 0` in the codebase.

Reply:
better handle the currentObject==nil case independently from the error case.

Reply:
@sttts Thanks! PTAL